### PR TITLE
Add optional team parameter to GetTeamInfo function calls

### PR DESCRIPTION
**Pull request by ed-wp**
Solves: https://github.com/slack-go/slack/issues/980

##### Pull Request Guidelines

These are recommendations for pull requests.
They are strictly guidelines to help manage expectations.

##### PR preparation
- [x] Run `make pr-prep` from the root of the repository to run formatting, linting and tests.

##### Should this be an issue instead
- [ ] is it a convenience method? (no new functionality, streamlines some use case)
- [ ] exposes a previously private type, const, method, etc.
- [ ] is it application specific (caching, retry logic, rate limiting, etc)
- [ ] is it performance related.

##### API changes

Since API changes have to be maintained they undergo a more detailed review and are more likely to require changes.

- [no tests, if you're adding to the API include at least a single test of the happy case.
- If you can accomplish your goal without changing the API, then do so.
- dependency changes. updates are okay. adding/removing need justification.

###### Examples of API changes that do not meet guidelines:
- in library cache for users. caches are use case specific.
- Convenience methods for Sending Messages, update, post, ephemeral, etc. consider opening an issue instead.

### CODE REVIEW

File: team.go [hunk 1]
```diff
@@ -96,15 +96,22 @@ func (api *Client) accessLogsRequest(ctx context.Context, path string, values ur
 }
 
 // GetTeamInfo gets the Team Information of the user
-func (api *Client) GetTeamInfo() (*TeamInfo, error) {
-	return api.GetTeamInfoContext(context.Background())
+// if the team is empty (""), will return information about the current team.
+// https://api.slack.com/methods/team.info
+func (api *Client) GetTeamInfo(team string) (*TeamInfo, error) {
```
Comment (ShamanR) [inline team.go:101]: Backward compatibility issue.

**Review by ShamanR — COMMENTED** [review]
sould we test  emty team case?

File: team.go (hunk 1 above)
Comment (kanata2) [reply team.go:101]: I'm having trouble with only one optional parameter...
I've come up with a few ideas, which one do you think is better?

### other idea 1
How about adding `GetAnotherTeamInfo(team string)` method .

```go
// Get your own team information.
info, err := api.GetTeamInfo()

// Get other team information.
info, err := api.GetOtherTeamInfo(teamID)
```

### other idea 2
Use functional option pattern.

```go
// Get your own team information.
info, err := api.GetTeamInfo()

// Get other team information.
info, err := api.GetTeamInfo(slack.GetTeamInfoOptionTeamID(teamID))
```

Comment (ed-wp) [reply team.go:101]: I feel like even though this is not backwards compatible, it follows how the slack API actually operates and makes it less confusing. 

From [Slack API Docs](https://api.slack.com/methods/team.info):
```
Team to get info on, if omitted, will return information about the current team. Will only return team that the authenticated token is allowed to see through external shared channels
```
I think that other option 2 is closer to the APIs definition, but not nearly as close as my proposal.

Thoughts?

Comment (kanata2) [reply team.go:101]: > I think that other option 2 is closer to the APIs definition, but not nearly as close as my proposal.

The parameter `team` is an optional in Slack while it is required in our `GetTeamInfo` API with your proposal, so I think it is not so good design.
The difference in idea 2 is that it keeps `team` parameter optional.

**Comment by kanata2** [comment]
#1118 is merged.

**Pull request closed without merging**